### PR TITLE
chore: sync, dependencies, docs

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

## Commits

- refactor: CI.yaml - use comp.action actions/gather-inputs chore: update dependencies; doc: AI CONVENTIONS.md
- deps: Bump the minor-and-patch group with 1 update
- refactor: CI.yaml - use comp.action actions/gather-inputs chore: update dependencies; doc: AI CONVENTIONS.md

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: .github/CONVENTIONS.md [hunk 1]
```diff
@@ -1,81 +1,239 @@
...
+
+> [!NOTE]
+> This file contains common conventions for contributing to the *vm2* repos targeting both humans and the AI coding systems. It is **copied to each repo's `.github/` directory**. The content of this file should be relatively stable and constant across repos. However, it may evolve over time as we learn and grow. Changes will be discussed, possibly leading to change in the SoT file at `$VM2_REPOS/vm2.Templates/templates/AddNewPackage/content/.github/CONVENTIONS.md`, and then propagate it to all repos.
+>
+> It is quite possible that some projects and solutions are "special" (e.g. vm2.DevOps) and require extra conventions and/or deviation from the shared conventions. `diff-shared.sh` allows for that: merge the source into the special target, instead of manual copy/paste. For more details about the `diff-shared.sh` script, clone the repo vm2.DevOps if you haven't done so and see the [tool's documentation](../vm2.DevOps/docs/diff-shared.md). Please, refrain from changing local copies of SoT files manually and thus deviating from the conventions in individual repos - use the tool.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md:43]: no

File: .github/CONVENTIONS.md [hunk 2]
```diff
@@ -1,81 +1,239 @@
...
+- Files which content is only partially shared are marked with a double asterisk **\*\***. If any of those change, **the sharing tool will display the differences and ask if you want to ignore the new content, copy the new content (overwriting the existing file), or merge the new content with the existing file** (both the diff tool and the merge tool are configurable in the script or use the git defaults - see the documentation for details)
+- Files that are not shared and should be maintained separately in each repo are not marked
+- These conventions are implemented by the `diff-shared.sh` script, which is used to propagate changes from the source-of-truth files in the template content folder to the target repos. The script can be configured to specify which files are shared and how to handle changes to them. For even more nuanced and flexible control, it defines 6 different actions for handling changes to shared files:
+  - **`ignore`**: do not update the target file, keep it as is
+  - **`merge_or_copy`**: ask the user to choose between ignoring, merging or copying the new content over the existing the new content.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md]: fixed

File: .github/CONVENTIONS.md [hunk 3]
```diff
@@ -1,81 +1,239 @@
...
+  - **`ask_to_merge`**: ask the user if they want to merge the new content with the existing file; if they choose not to merge, do not update the target file
+  - **`merge`**: open the merge utility without asking the user to merge the new content with the existing file, preserving both the shared content and the repo-specific content
+  - **`ask_to_copy`**: ask the user if they want to copy the new content over the existing file; if they choose not to copy, do not update the target file
+  - **`copy`**: copy the new content over the existing file (overwriting it) without asking
+
+  By default all documents marked with **\*** are set to `copy`, and all documents marked with **\*\*** are set to `merge_or_copy`.
+
+  For more details on how to use the `diff-shared.sh` script, see the [tool's documentation](../vm2.DevOps/docs/diff-shared.md).
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md:102]: no

File: .github/CONVENTIONS.md [hunk 4]
```diff
@@ -1,81 +1,239 @@
...
-- No curly braces for single-line blocks unless they improve readability.
-- `#region` / `#endregion` acceptable for logical grouping in larger files.
-- EBNF (ISO 14977) for grammar definitions: `=` definitions, `,` concatenation,
-  `;` rule terminator, `[ ]` optional, `|` alternation, `"..."` terminals.
+  - `test/` — test projects. The stack is: **xUnit, FluentAssertions, NSubstitute, MTP v2, coverage**
+  - `benchmarks/` — **BenchmarkDotNet** projects (desirable)
+  - `examples/` — usage examples (desirable). Prefer single-file programs for simplicity, but multi-file projects are acceptable if the example is complex enough to warrant it.
+  - `docs/` — documentation (optional, in addition to README.md, e.g. blogs, design docs, etc.)
+  - `.github/workflows/` — **GitHub Actions CI/CD**:
+    - `CI.yaml` kicks-in the **inputs validation, build, test, benchmark, and package** shared workflows
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md]: fixed

File: .github/CONVENTIONS.md [hunk 5]
```diff
@@ -1,81 +1,239 @@
...
+    - `Prerelease.yaml` for **prerelease workflows**: computes the pre-release version, e.g. `1.2.0-preview.3`; tags the main branch with a tag like `v1.2.0-preview.3`; builds pre-release package(s); **publishes the pre-release packages to GitHub Packages or NuGet.org**; **updates CHANGELOG.md** using cliff-git. Triggered by a successful merge of a pull request
+    - `Release.yaml` for **release workflows**: computes the release version, e.g. `1.2.0`; tags the main branch with a tag like `v1.2.0`; builds release package(s); **publishes the pre-release packages to GitHub Packages or NuGet.org**; **updates CHANGELOG.md** using cliff-git. Triggered manually when we want to cut a release, usually after a successful prerelease validation
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md]: fixed

[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md]: fixed

File: .github/CONVENTIONS.md [hunk 6]
```diff
@@ -1,81 +1,239 @@
...
-- Exceptions for unrecoverable failures (e.g. `ArgumentException`).
 - `Try...` patterns preferred over broad exception-based control flow.
+- **Domain-specific exceptions** for business rule violations
+- **Never swallow exceptions** — log or rethrow
+- **Never log sensitive data** (PII, secrets)
+- **Use logger scopes** for contextual information, not string concatenation or interpolation in log messages
+- **Prefer `ILogger<T>`** with structured logging over static loggers or string-based logging
+- In services with external dependencies, prefer **circuit breakers and retries** over exceptions for transient faults
+- In services use **health checks and monitoring** to detect and respond to failures instead of relying on exceptions for observability
+- In services for distributed systems **use Open Telemetry** for distributed tracing and metrics to understand system behavior and failures instead of relying on exceptions alone
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md]: fixed

File: .github/workflows/CI.yaml [hunk 7]
```diff
@@ -90,106 +90,27 @@ jobs:
...
-              dotnet_version \
-              configuration \
-              preprocessor_symbols \
-              min_coverage_pct \
-              max_regression_pct \
-              minver_tag_prefix \
-              minver_prerelease_id \
-              reset_benchmark_thresholds \
-              skip_push
+        uses: vmelamed/vm2.DevOps/.github/actions/gather-inputs@main
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/workflows/CI.yaml:95]: this is mine but will do some day
